src/init.cpp

```diff
@@ -1438,6 +1438,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         options.prune = node::fPruneMode;
         options.check_blocks = args.GetIntArg("-checkblocks", DEFAULT_CHECKBLOCKS);
         options.check_level = args.GetIntArg("-checklevel", DEFAULT_CHECKLEVEL);
+        options.fail_on_insufficient_dbcache = args.IsArgSet("-checkblocks") || args.IsArgSet("-checklevel");
```

Comment:
42402bffda6643346cb2b51c9e85b389f3d19068: Any reason for this bool? The `DEFAULT_CHECKLEVEL` is `3`, presumably because `4` would almost never work anyway. And with `3` the check can't fail due to insufficient cache. Seems better to remove this bool and "replace" it with `true`?

Reply:
Ah, I see previously it would silently pass, but in this pull you changed it to an error

Reply:
I wonder if this requires a release note, or alternatively if the patch would be smaller and easier to review if it only fixed the level 4 issue and left level 3 as-is for now?

Reply:
I added a release note, for the rest see the top-level comment below.